src/scripts/GameController.ts

```diff
@@ -159,6 +191,26 @@ class GameController {
                             }
                         }
                     }
+                    if ($farmsModal.data('bs.modal')?._isShown) {
+                        if (e.code == 'KeyS') {
+                            FarmController.selectedShovel() ? FarmController.selectedShovel(false) : FarmController.selectedShovel(true);
+                        }
+                    } else if ($undergroundModal.data('bs.modal')?._isShown) {
+                        if (e.code == 'KeyH') {
+                            Mine.toolSelected(Mine.Tool.Hammer);
```

Comment:
This doesn't seem to be working for me, it's just bringing up the Hatchery when I try.

Reply:
Yeah, did check that after I added the Hatchery. Moved that key into a check in the H case above it.